adafruit_connection_manager.py

```diff
@@ -224,23 +213,24 @@ def _get_connected_socket(  # pylint: disable=too-many-arguments
         is_ssl: bool,
         ssl_context: Optional[SSLContextType] = None,
     ):
-        try:
-            socket = self._socket_pool.socket(addr_info[0], addr_info[1])
-        except (OSError, RuntimeError) as exc:
-            return exc
+
+        socket = self._socket_pool.socket(addr_info[0], addr_info[1])
 
         if is_ssl:
             socket = ssl_context.wrap_socket(socket, server_hostname=host)
             connect_host = host
         else:
             connect_host = addr_info[-1][0]
-        socket.settimeout(timeout)  # socket read timeout
+
+        # Set socket read and connect timeout.
+        socket.settimeout(timeout)
 
         try:
             socket.connect((connect_host, port))
-        except (MemoryError, OSError) as exc:
+        except (MemoryError, OSError):
```

Comment:
Should this be any exception?

Reply:
This is deliberately not that, because I was thinking that these are the only errors that should be retried. Maybe there are some others that should be retried also. But some errors are unrecoverable and so I don't want to catch them: I want the user program to fail (or catch them).